pkg/controller/core/indexer/indexer.go

```diff
@@ -94,15 +94,13 @@ func IndexWorkloadAdmitted(obj client.Object) []string {
 }
 
 func IndexWorkloadRuntimeClass(obj client.Object) []string {
```

Comment:
We probably should leave this indexer and add an additional one for PodTemplate.